src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -483,7 +551,8 @@ public void replaceAction(Action a) {
     }
 
     @Override public SCMTrigger getSCMTrigger() {
-        return triggers.get(SCMTrigger.class);
+        // TODO: This seems fairly weird to me and I'm not 100% sure why it works.
+        return (SCMTrigger) getTriggers().get(getDescriptorByName("SCMTrigger"));
```

Comment:
Just search the list for `instanceof SCMTrigger`? There is a helper function somewhere in `Util` IIRC.